### DISCUSSION TITLE
gradunwarp: gradient field inhomogenities unwarping

### DISCUSSION THREAD

**Discussion by bpinsard** [nipreps/fmriprep#2818] [Ideas]
I am starting to investigate the feasibility to implement [gradunwarp](https://github.com/Washington-University/gradunwarp) step in fMRIPrep and list the changes to be made in the different packages.
- [ ] niworkflows
  - [ ] Create an interface for gradunwarp estimation
  - [ ] Implement a single-step interpolation that combines gradunwarp, HMC, and SDC
    - [ ] Can the gradunwarp field  be applied by AntsApplyTransforms or does it require a conversion (nitransforms)
  - [ ] BIDS/automation: detect 'ND' in ImageType of all images to determine if they need offline distortion correction or not
- [ ] smriprep: 
  - [ ] add option to pass Siemens coefficients 
  - [ ] unwarp T1w/T2w/Flair images prior to robust template creation or longitudinal pipeline
- [ ] fmriprep:
  - [ ] add option to pass Siemens coefficients 
  - [ ] unwarp EPIs fmap passed to sdcflows (or should that be included in sdcflows)
  - [ ] unwarp bold in the single-step transform implemented in niworkflow. 

Am I missing anything?

**Comment by effigies** [comment]
This looks right to me. Please see also https://github.com/nipreps/fmriprep/issues/1550 for comments from those more knowledgeable than myself.
